File: python/nvstrings/tests/test_case.py
# Copyright (c) 2018-2019, NVIDIA CORPORATION.

import nvstrings
from utils import assert_eq


def test_lower():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.lower()
    expected = ['abc', 'def', None, 'jll']
    assert_eq(got, expected)


def test_upper():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.upper()
    expected = ['ABC', 'DEF', None, 'JLL']
    assert_eq(got, expected)


def test_swapcase():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.swapcase()
    expected = ['ABC', 'dEF', None, 'JlL']
    assert_eq(got, expected)


def test_capitalize():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.capitalize()
    expected = ['Abc', 'Def', None, 'Jll']
    assert_eq(got, expected)


def test_title():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.title()
    expected = ['Abc', 'Def', None, 'Jll']
    assert_eq(got, expected)


def test_islower():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.islower()
    expected = [True, False, None, False]
    assert_eq(got, expected)


def test_isupper():
    strs = nvstrings.to_device(["abc", "Def", None, "jLl"])
    got = strs.isupper()
    expected = [False, False, None, False]
    assert_eq(got, expected)

File: cpp/custrings/tests/test_case.cpp
#include <gtest/gtest.h>
#include <vector>

#include "nvstrings/NVStrings.h"

#include "./utils.h"

struct TestCase : public GdfTest{};

std::vector<const char*> hstrs{ "Examples aBc", "thesé", nullptr, "ARE THE", "tést strings", "" };

TEST_F(TestCase, ToLower)
{
    NVStrings* strs = NVStrings::create_from_array(hstrs.data(),hstrs.size());
    NVStrings* got = strs->lower();
    const char* expected[] = { "examples abc", "thesé", nullptr, "are the", "tést strings", "" };
    EXPECT_TRUE( verify_strings(got,expected));
    NVStrings::destroy(got);
    NVStrings::destroy(strs);
}

TEST_F(TestCase, ToUpper)
{
    NVStrings* strs = NVStrings::create_from_array(hstrs.data(),hstrs.size());
    NVStrings* got = strs->upper();
    const char* expected[] = { "EXAMPLES ABC", "THESÉ", nullptr, "ARE THE", "TÉST STRINGS", "" };
    EXPECT_TRUE( verify_strings(got,expected));
    NVStrings::destroy(got);
    NVStrings::destroy(strs);
}

TEST_F(TestCase, Swapcase)
{
    NVStrings* strs = NVStrings::create_from_array(hstrs.data(),hstrs.size());
    NVStrings* got = strs->swapcase();
    const char* expected[] = { "eXAMPLES AbC", "THESÉ", nullptr, "are the", "TÉST STRINGS", "" };
    EXPECT_TRUE( verify_strings(got,expected));
    NVStrings::destroy(got);
    NVStrings::destroy(strs);
}

TEST_F(TestCase, Capitalize)
{
    NVStrings* strs = NVStrings::create_from_array(hstrs.data(),hstrs.size());
    NVStrings* got = strs->capitalize();
    const char* expected[] = { "Examples abc", "Thesé", nullptr, "Are the", "Tést strings", "" };
    EXPECT_TRUE( verify_strings(got,expected));
    NVStrings::destroy(got);
    NVStrings::destroy(strs);
}

TEST_F(TestCase, Title)
{
    NVStrings* strs = NVStrings::create_from_array(hstrs.data(),hstrs.size());
    NVStrings* got = strs->title();
    const char* expected[] = { "Examples Abc", "Thesé", nullptr, "Are The", "Tést Strings", "" };
    EXPECT_TRUE( verify_strings(got,expected));
    NVStrings::destroy(got);
    NVStrings::destroy(strs);
}
